src/algorithms/generate.rs

```diff
@@ -186,4 +200,19 @@ mod tests {
     key_generation!(key_generation_multi_8_576, 8, 576);
     // TODO: reenable, currently slow
     // key_generation!(key_generation_multi_16_1024, 16, 1024);
+
+    #[test]
+    fn test_log_approx() {
+        let mut rng = ChaCha8Rng::from_seed([42; 32]);
+
+        for i in 0..100 {
+            println!("round {i}");
+            let prime_limit: f64 = rng.gen();
+            let a = logf(prime_limit);
+            let b = logf_approx(prime_limit as f32);
+
+            let diff = a - b as f64;
+            assert!(diff < 0.001, "{} != {}", a, b);
```

Comment:
What is this supposed to test? `logf` calls `logf_approx`, so this currently tests whether casting from `f32` to `f64` does not result in big difference?

Reply:
Maybe test against [`ln`](https://doc.rust-lang.org/std/primitive.f64.html#method.ln) from the standard library instead?

Reply:
logf calls ln when compiled under std

Reply:
I missed that from just reading the diff, the test makes sense now!